src/OSPSuite.Core/Domain/Services/CalculationMethodTask.cs

```diff
@@ -144,30 +131,7 @@ private static Neighborhood neighborhoodAncestorFor(IEntity entity)
          return neighborhoodAncestorFor(entity.ParentContainer);
       }
 
-      private bool formulasAreTheSameForParameter(IParameter originalParameter, IFormula calculationMethodFormula, string moleculeName)
-      {
-         var previousFormula = originalParameter.Formula;
-         //needs to use the parameter in order to keep the hierarchy. Hence we set the formula in the parameter
-         originalParameter.Formula = _formulaMapper.MapFrom(calculationMethodFormula, _simulationBuilder);
-
-         //check  if the formula set are the same. it is necessary to replace keywords before doing that
-         replaceKeyWordsIn(originalParameter, moleculeName);
-
-         try
-         {
-            return _formulaTask.FormulasAreTheSame(originalParameter.Formula, previousFormula);
-         }
-         finally
-         {
-            //reset the origianl formula in any case
-            originalParameter.Formula = previousFormula;
-         }
-      }
-
-      private bool parameterIsNotBlackBoxParameter(IParameter parameter)
-      {
-         return !_allBlackBoxParameters.Contains(parameter);
-      }
+      private bool parameterIsNotBlackBoxParameter(IParameter parameter) => !_allBlackBoxParameters.Contains(parameter);
```

Comment:
I wonder if just checking for `parameter.Formula.IsBlackBox()` here is not more performant than caching all black box parameters from the model into _allBlackBoxParameters and checking if a parameter is in that list every time
(Because when caching, `parameter.Formula.IsBlackBox()` is called for **every model parameter** anyway; and the function `parameterIsNotBlackBoxParameter()` is called only once per parameter and only for the _calculation methods_ parameters)

Apart from that looks good for me.